documentation/design/user-stories/shared-infra.adoc

```diff
@@ -229,75 +280,297 @@ metadata:
+which namespaces it can be referenced by.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingAddressPlan
+metadata:
+  name: myplan
+  namespace: planspace
+spec:
+  # Selects which namespaces this tenant plans should be available for
+  selector:
+    # Restrict to specific namespaces
+    namespaces:
+     - ns1
+     - ns2
+    # Restrict to namespaces matching labels
+    namespaceSelector:
+      matchLabels:
+        environment: production
+```
+
+
+=== Specify default plans for messaging tenants
+
+Specify the following in the `MessagingInfra` to select which plan should be the default for that infra. By default,
+an unrestricted plan will be used as default.
+
+* If the reference does not resolve, the infra status will contain a message describing the issue.
+* A warning status will be set if the union of the `MessagingInfra` and the `MessagingPlan` selector is a disjoint set.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingInfra
+metadata:
+  name: myinfra
+  namespace: infraspace
+spec:
+  defaultPlanRef:
+    name: myplan
+    namespace: planspace
+```
 
 === Specify default address plans for messaging tenants
 
-=== CRUDLW an instance of messaging infrastructure shared between multiple messaging tenants
+TODO
+
+=== Specify default endpoints and cert providers to be created for messaging tenants
+
+Default endpoint configuration is specified in the `MessagingPlan` under the `default` field.
+
+* The controller creates one `MessagingEndpoint` for each entry in the plan list
+* The schema for each entry corresponds to the `spec` object in `MessagingEndpoint` with the following exceptions:
+  * The `host` field is ignored and set to defaults
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingPlan
+metadata:
+  name: myplan
+  namespace: planspace
+spec:
+  default:
+    endpoints:
+     - route: {}
+       protocols:
+       - AMQPS
+       tls:
+         openshift: {}
+     - cluster: {}
+        # Define which protocols are supported by the default endpoint
+       protocols:
+       - AMQP
+```
+
+=== Restrict which endpoint types and cert providers can be used for messaging tenants
+
+Restricting allowed endpoint types and cert providers allow the service admin to control
+exposure of messaging endpoints in a cluster. The `restrict.endpoints` field is used to restrict
+which endpoint types and which protocols are allowed.
+
+* `MessagingEndpoint` resources for tenants referencing this plan must exactly match one of the entries.
+* The schema for each entry corresponds to the `spec` object in `MessagingEndpoint` with the following exceptions:
+  * The `host` field is ignored and set to defaults
+
+The following example will enforce `route` endpoints to use a specific certificate and require `cluster` endpoints to use unencrypted protocols.
+
+```
+apiVersion: enmasse.io/v1beta2
+kind: MessagingPlan
+metadata:
+  name: myplan
+  namespace: planspace
+spec:
+  restrict:
+    endpoints:
```

Comment:
I am wondering if there is a more expressive way to let the user encode restrictions.  It reminds me of schematron's niche  back in the XML world (Pangea :) ).